### ISSUE TITLE
Accessibility: File/folder select checkbox has invalid label

### ISSUE THREAD

**Issue by ChloeSartorelli** [silverstripe/silverstripe-asset-admin#1594]
[WCAG 2.2 A 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html) 1.3.1 2.4.6

The checkbox used to select files and folders in asset admin has several problems:

1. The label has no text associated with it, so screenreaders won't even have anything to read out
2. If you add text (even if visually hidden) to the label, it makes the checkbox stop working
3. The checkbox itself has a `title` attribute that just says "Select", but doesn't indicate _what_ you're selecting. I assume this was intended to be used as a tooltip but it doesn't display when hovering over the checkbox anyway so it's not helpful.

## Notes

Check with testing and/or UX team if the tooltip should be implemented or not.

## Acceptance criteria

- [ ] The `title` attribute is removed from the checkbox
- [ ] The label is given appropriate visually hidden text, such as "Select file named '<file name here>'" or "Select folder named '<folder name here>'" _**OR**_ is replaced with an `aria-label` attribute on the checkbox
- [ ] If a tooltip displaying the label text on hover is desirable, implement it - but make sure screen readers don't repeat the same text twice.
- [ ] Selecting files and folders with both mouse and keyboard still works as expected
- [ ] Screen reader experience is not negatively impacted
- [ ] The changelog clearly calls out any changes to markup selectors so people can adjust their custom css/js as necessary
